Nodejs/Product/Npm/PackageComparer.cs

```diff
@@ -30,4 +30,22 @@ public class PackageComparer : IComparer<IPackage> {
             return x.Name.CompareTo(y.Name);
         }
     }
+
+    public class PackageEqualityComparer : EqualityComparer<IPackage> {
+        public override bool Equals(IPackage p1, IPackage p2) {
+            return p1.Name == p2.Name
+                && p1.Version == p2.Version
+                && p1.IsBundledDependency == p2.IsBundledDependency
+                && p1.IsDevDependency == p2.IsDevDependency
+                && p1.IsListedInParentPackageJson == p2.IsListedInParentPackageJson
+                && p1.IsMissing == p2.IsMissing
+                && p1.IsOptionalDependency == p2.IsOptionalDependency;
+        }
+
+        public override int GetHashCode(IPackage obj) {
+            if (obj.Name == null || obj.Version == null)
+                return obj.GetHashCode();
+            return obj.Name.GetHashCode() ^ obj.Version.GetHashCode();
+        }
+    }
```

Comment:
Ahh, doesn't look like anything got pulled from ReloadHierarchy then. Additionally, there's a PackageComparer above - can you take a look at how that's being used see whether it makes sense to consolidate anything?